fixer/fixer.go

```diff
@@ -0,0 +1,82 @@
+package fixer
+
+import (
+	"go/ast"
+	"go/parser"
+	"path/filepath"
+
+	"github.com/go-logr/logr"
+	"github.com/thockin/klog-to-logr/importer"
+)
+
+// FileInfo represents the information needed to process some file
+// to apply a fix.
+type FileInfo struct {
+	Name string
+	Package *importer.PackageInfo
+	AST *ast.File
+}
+
+// Fix represents some fix to be applied to a given file.
+type Fix struct {
+	Name string
+	Execute func(FileInfo, importer.Loader, logr.Logger) bool
+	Description string
+}
+
+// Fixer executes fixes against packages
+type Fixer struct {
+	Log logr.Logger
+	Fixes []Fix
+	Loader importer.Loader
+
+	HandleFix func(FileInfo) error
+}
+
+func (f *Fixer) FixPackage(pkg *importer.PackageInfo) error {
+	pkgLog := f.Log.WithValues("package", pkg.BuildInfo.ImportPath)
+	pkgLog.V(1).Info("applying fixes", "file count", len(pkg.Files))
```

Comment:
Shouldn't logr keys be single tokens by convention?

Reply:
I've been using space-separated by convention elsewhere.  It doesn't make a whole ton of difference.